homework/index.js

```diff
@@ -29,23 +31,141 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function addRow(table, title, repoText) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, {
+      text: title,
+    });
+    createAndAppend('td', tr, { text: repoText });
+    return tr;
+  }
+
+  function renderRepoDetails(repo, parent) {
+    const date = new Date(repo.updated_at);
+    const table = createAndAppend('table', parent);
+    const firstRow = addRow(table, 'Repository:', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: '_blank',
+    });
+
+    addRow(table, 'Description:', repo.description, 'td');
+    addRow(table, 'Forks:', repo.forks, 'td');
+    addRow(table, 'Updated:', date.toLocaleDateString(), 'td');
+  }
+
+  function createOption(repo, parent, index) {
+    createAndAppend('option', parent, {
+      text: repo.name,
+      value: index,
+    });
+  }
+
+  function selectedIndex(selection, repositories, parent) {
+    const repo = repositories[selection];
+    renderRepoDetails(repo, parent);
+  }
+
+  function createContributorsSection(url, parent) {
+    fetchJSON(url)
+      .then(contributes => {
+        contributes.forEach(contribute => {
+          const li = createAndAppend('li', parent);
+          createAndAppend('img', li, {
+            src: contribute.avatar_url,
+            class: 'avatar',
+          });
+          createAndAppend('a', li, {
+            text: contribute.login,
+            href: contribute.html_url,
+            target: '_blank',
+            class: 'cont-name',
+          });
+          createAndAppend('p', li, {
+            text: contribute.contributions,
+          });
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', parent, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      });
+  }
+
+  function firstScreen(repos, repoParent, contParent, index) {
+    renderRepoDetails(repos[index], repoParent);
+    createContributorsSection(repos[index].contributors_url, contParent);
+  }
+
+  function changeSelection(select, repos, repoParent, contParent) {
+    const selection = select.value;
+    const contUrl = repos[selection].contributors_url;
+    selectedIndex(selection, repos, repoParent);
+    createContributorsSection(contUrl, contParent);
   }
```

Comment:
You can simplify this function to:

```js
function changeSelection(repo, repoParent, contParent) {
  repoParent.innerHTML = '';
  contParent.innerHTML = '';
  renderRepoDetails(repo, repoParent);
  createContributorsSection(repo.contributors_url, contParent);
}
```

You just need to pass the selected repo as the first parameter. For the first screen this would be the first repo in the repos array. And otherwise, it would be the selected repo from the select box.

Reply:
@remarcmij it is possible for my fault but it didn't work. The first parameter worked, I already do it with your hint but selection parameters didn't work. Result: All contributors shows. I have to add selections function in changeselection function.

Reply:
Got it  👍